extensions-core/s3-extensions/src/main/java/org/apache/druid/data/input/s3/S3InputSource.java

```diff
@@ -98,16 +109,49 @@ public S3InputSource(
       @JsonProperty("objects") @Nullable List<CloudObjectLocation> objects,
       @JsonProperty("filter") @Nullable String filter,
       @JsonProperty("properties") @Nullable S3InputSourceConfig s3InputSourceConfig,
+      @JsonProperty("proxyConfig") @Nullable AWSProxyConfig awsProxyConfig,
+      @JsonProperty("endpointConfig") @Nullable AWSEndpointConfig awsEndpointConfig,
+      @JsonProperty("clientConfig") @Nullable AWSClientConfig awsClientConfig,
```

Comment:
Reusing the Config objects here so that we don't duplicate the properties. We could push these under `S3InputSourceConfig` itself if needed.